### DISCUSSION TITLE
Is there a way to check what kubernetes resources will be altered on change (Similar to terraform plan)

### DISCUSSION THREAD

**Discussion by preston-willis** [fluxcd/flux2#4580] [Proposals]
It would be cool to have a pre-commit hook at shows a summary of the changes that will be made given manifest changes, any clue how this could be done / if it exists?

**Comment by stefanprodan** [comment]
Yes, you can use the `flux diff` command that will connect to the cluster and print the changes as they will happen once you commit them to the repo.

**Reply by preston-willis** [reply]
So I could use this in a git pipeline and connect to my cluster? Does it just check the files in the PWD and compare to the cluster state?

**Reply by preston-willis** [reply]
Looking into how to do this, how can I connect to my cluster from a bitbucket pipeline? Not sure how to grab the TLS cert
